### PR TITLE
Saching/sharepoint support

### PR DESCRIPTION
**Pull request by sachingupta**
Adding regex support for valid host domains.
added sharepoint domain in valid host regex

### CODE REVIEW

File: src/MicrosoftTeams.ts [hunk 1]
```diff
@@ -21,14 +21,26 @@ namespace microsoftTeams {
 
   const version = "1.2";
 
-  const validOrigins = [
-    "https://teams.microsoft.com",
-    "https://teams.microsoft.us",
-    "https://int.teams.microsoft.com",
-    "https://devspaces.skype.com",
-    "https://ssauth.skype.com",
+  const validHostRegEx = new RegExp(
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: NIT: Consider making this a little easier to extend by creating the regex dynamically in code out of an array

File: src/MicrosoftTeams.ts [hunk 2]
```diff
@@ -21,14 +21,26 @@ namespace microsoftTeams {
...
 
-  const validOrigins = [
-    "https://teams.microsoft.com",
-    "https://teams.microsoft.us",
-    "https://int.teams.microsoft.com",
-    "https://devspaces.skype.com",
-    "https://ssauth.skype.com",
-    "http://dev.local" // local development
-  ];
+  const validHostRegEx = new RegExp(
```
Comment (aosolis) [inline src/MicrosoftTeams.ts]: Can we name this back to validOriginsRegExp, since it's actually checking the origin, not just the host?

File: src/MicrosoftTeams.ts [hunk 3]
```diff
@@ -21,14 +21,26 @@ namespace microsoftTeams {
...
+    "|" +
+    "https://devspaces.skype.com" +
+    "|" +
+    "https://ssauth.skype.com" +
+    "|" +
+    "http://dev.local" + // local development
+    "|" +
+    "https://msft.spoppe.com" +
+    "|" +
+    "http*.*sharepoint.com" +
```
Comment (aosolis) [inline src/MicrosoftTeams.ts]: This looks wrong. I think this clause of the regexp means
    `"htt" + (zero or more "p"s) + (zero or more of any characters) + "sharepoint" + (any character) + com`

which would match the following:
- htt://x.phishing-site-sharepoint.com
- httpppppp://x.malicious-sharepoint.com
- httpppppp://x.malicious-sharepoint-com
- httpmalicious-sharepointxcom

File: src/MicrosoftTeams.ts (hunk 2 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: sure

File: src/MicrosoftTeams.ts (hunk 3 above)
Comment (ydogandjiev) [reply src/MicrosoftTeams.ts]: Consider re-using the logic we have for domain validation in extensionTabService.ts.

File: src/MicrosoftTeams.ts [hunk 4]
```diff
@@ -1,3 +1,13 @@
+declare interface String {
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Hmm... Why are these needed?

File: test/MicrosoftTeams.spec.ts [hunk 5]
```diff
@@ -169,20 +181,63 @@ describe("MicrosoftTeams", () => {
...
+          id: getContextMessage.id,
+          args: [
+            {
+              groupId: "someMaliciousValue"
+            }
+          ]
+        } as MessageResponse
+      } as MessageEvent);
+
+      expect(callbackCalled).toBe(true);
```
Comment (aosolis) [inline test/MicrosoftTeams.spec.ts]: What resets callbackCalled to false for each item in supportedDomains? I think this ends up checking only that the callback was called for at least *one* of the domains, not each one.

File: test/MicrosoftTeams.spec.ts [hunk 6]
```diff
@@ -169,20 +181,63 @@ describe("MicrosoftTeams", () => {
...
+          id: getContextMessage.id,
+          args: [
+            {
+              groupId: "someMaliciousValue"
+            }
+          ]
+        } as MessageResponse
+      } as MessageEvent);
+
+      expect(callbackCalled).toBe(false);
```
Comment (aosolis) [inline test/MicrosoftTeams.spec.ts]: Similarly here it checks that the callback was never called (which is correct), but should it fail and a callback does end up getting called, we don't keep enough information to identify which one was missed.

File: src/MicrosoftTeams.ts [hunk 7]
```diff
@@ -21,15 +31,45 @@ namespace microsoftTeams {
...
   const validOrigins = [
     "https://teams.microsoft.com",
     "https://teams.microsoft.us",
     "https://int.teams.microsoft.com",
     "https://devspaces.skype.com",
     "https://ssauth.skype.com",
-    "http://dev.local" // local development
+    "http://dev.local", // local development
+    "https://msft.spoppe.com",
+    "*.sharepoint.com",
```
Comment (aosolis) [inline src/MicrosoftTeams.ts]: The mix of explicit origins and wildcarded hostnames makes this hard to maintain-- someone will be tempted to add an entry like `https://*.teams.microsoft.com` which will not work as expected.

File: test/MicrosoftTeams.spec.ts [hunk 8]
```diff
@@ -169,20 +181,63 @@ describe("MicrosoftTeams", () => {
...
+  it("should allow messages from supported domains", () => {
+    initializeWithContext("content");
+    const supportedDomains = [
+      "https://teams.microsoft.com",
+      "https://teams.microsoft.us",
+      "https://int.teams.microsoft.com",
+      "https://devspaces.skype.com",
+      "http://dev.local",
+      "https://microsoft.sharepoint.com",
+      "https://a.b.sharepoint.com",
```
Comment (aosolis) [inline test/MicrosoftTeams.spec.ts]: Do these pass? I think "*" in the pattern matches *exactly* one subdomain.

File: test/MicrosoftTeams.spec.ts (hunk 5 above)
Comment (sachingupta) [reply test/MicrosoftTeams.spec.ts]: thanks i am fixing Unit tests logic

File: test/MicrosoftTeams.spec.ts (hunk 8 above)
Comment (sachingupta) [reply test/MicrosoftTeams.spec.ts]: Looks like the way i have setup Unit test is wrong. i am working on fixing it.

File: src/MicrosoftTeams.ts (hunk 3 above)
Comment (sachingupta) [reply src/MicrosoftTeams.ts]: updated with domain validation logic in extensiontabService
